json-rpc/docs/type_transaction.md

```diff
@@ -18,7 +18,9 @@ A transaction on the blockchain.
 | vm_status                 | [VMStatus](#type-vmstatus)               | The returned status of the transaction after being processed by the VM                     |
 | gas_used                  | unsigned int64 | Amount of gas used by this transaction, to know how much you paid for the transaction, you need multiply it with your RawTransaction#gas_unit_price |
 
-Note: For the gas_used, internally within the VM we scale the gas units down by 1000 in order to allow granularity of costing for instruction, but without having to use floating point numbers, but we do round-up the gas used to the nearest "1" when we convert back out.
+Note:
+* For the gas_used, internally within the VM we scale the gas units down by 1000 in order to allow granularity of costing for instruction, but without having to use floating point numbers, but we do round-up the gas used to the nearest "1" when we convert back out.
+* Formula to create hash for a signed transaction before it is executed: hex-encode(sha3-256([]byte("LIBRA::Transaction")) + []byte {0} + signed transaction bytes) ([implementation example](https://github.com/libra/libra-client-sdk-go/blob/master/libratypes/hash.go#L27))
```

Comment:
what does the 0 represent?

Reply:
also maybe using (byte[]) like casting is easier to understand?

Reply:
0 represent Transaction first variant (enum) value UserTransaction

Reply:
oh yeah, I figured, but it wasn't defined here.

also the use of []byte seems very strange in a Rust repo :P